### PR TITLE
Fix array encoding in form data requests

### LINKED ISSUE #1713: samples html_form_post.js fails
K6 http.post method isn't correctly encoding a list of values assigned to a key in an object.

```javascript
const form_data = {
    comment: "Hello world!",
    topping: [
        'onion',
        'bacon',
        'cheese'
    ]
};
```
The simplest way to see this is to run the sample at 
https://github.com/loadimpact/k6/blob/master/samples/html_form_post.js and note that it fails.

The list of toppings should be passed as multiple values for the same key, but it is instead having the values stringified and passed as one value.

With this issue, it's difficult to pass multiple select check boxes with the same name or multi-selections in a select.

## Environment
- k6 version: k6 v0.28.0 (2020-09-24T14:33:59+0000/v0.28.0-0-gdee9c4ce, go1.14.9, linux/amd64)
- OS and version: Both windows 10 and Linux K6
- Docker version and image, if applicable:

## Expected Behavior

The topping values should be encoded as
topping=onion&topping=bacon&topping=cheese
and the sample should pass all checks.

## Actual Behavior

Test case fails because the topping values are encoded as (and URI encoded)
topping=[onion bacon cheese]

## Steps to Reproduce the Problem

1. Copy sample from https://github.com/loadimpact/k6/blob/master/samples/html_form_post.js locally or clone the git repository
2. k6 run samples/html_form_post.js
3. Observe test failure outcome

### PR DESCRIPTION
**Pull request by imiric**
This is a continuation of #1754, with an added check for the correct encoding and disabling of the `exhaustive` linter, since it doesn't seem like we can avoid the reflection.

Closes #1713

### CODE REVIEW

**Review by na-- — DISMISSED** [review]

File: js/modules/k6/http/request.go [hunk 1]
```diff
@@ -177,7 +177,16 @@ func (h *HTTP) parseRequest(
...
 			bodyQuery := make(url.Values, len(data))
 			for k, v := range data {
-				bodyQuery.Set(k, formatFormVal(v))
+				switch reflect.TypeOf(v).Kind() { //nolint:exhaustive
+				// handle json arrays in params
+				case reflect.Slice, reflect.Array:
+					s := reflect.ValueOf(v)
+					for i := 0; i < s.Len(); i++ {
+						bodyQuery.Add(k, formatFormVal(s.Index(i)))
+					}
```
Comment (MStoykov) [inline js/modules/k6/http/request.go:186]: Using a simple type switch and matching just `[]interface{}` works just as well. I have no idea how to make a js code that will give me a go array and apparently the test below doesn't do it ... so I am kind of for ditching the `reflect` ;)

Comment (na--) [reply js/modules/k6/http/request.go:186]: that was my objection in the original PR, but then I got this response and didn't verify for myself: https://github.com/k6io/k6/pull/1754#issuecomment-738773727

Comment (MStoykov) [reply js/modules/k6/http/request.go]: Well I did write the code .. and we do assert to `[]interface{}` in the batch parsing code:
https://github.com/k6io/k6/blob/b9a29eb7fc75b5d903f0c9e3a307ed80d06dacc5/js/modules/k6/http/request.go#L468-L474

Comment (imiric) [reply js/modules/k6/http/request.go]: Yeah, I assumed that comment to be true and didn't try a type assertion. Seems to work fine, see 0d48122b.

**Review by na-- — APPROVED** [review]

**Review by MStoykov — APPROVED** [review]
LGTM!

I would've used a `range` as it seems more idiomatic to me but this is fine as well

**Review by MStoykov — APPROVED** [review]

**Comment by imiric** [comment]
Apparently 2 approvals are required if a previous approval was dismissed? :-/

**Review by na-- — APPROVED** [review]

**Pull request merged**